File: backend/schemas/application.py
from pydantic import BaseModel


class SubmitApplicationRequest(BaseModel):
    job_listing_id: str
    cover_letter_id: str | None = None
    submission_method: str = "manual"
    notes: str | None = None


class UpdateStatusRequest(BaseModel):
    status: str


class ApplicationJobSummary(BaseModel):
    id: str
    title: str
    company: str
    location: str
    url: str


class ApplicationResponse(BaseModel):
    id: str
    user_id: str
    job_listing_id: str
    cover_letter_id: str | None = None
    status: str
    submission_method: str
    submitted_at: str
    last_updated_at: str
    notes: str | None = None
    job: ApplicationJobSummary | None = None

    class Config:
        from_attributes = True


class DashboardStats(BaseModel):
    total_applied: int
    interviews: int
    offers: int
    rejection_rate: float

File: packages/shared/src/types/application.ts
export type ApplicationStatus =
  | 'pending'
  | 'interview'
  | 'rejected'
  | 'offer'
  | 'withdrawn';

export interface Application {
  id: string;
  user_id: string;
  job_listing_id: string;
  cover_letter_id?: string;
  status: ApplicationStatus;
  submission_method: 'auto' | 'manual' | 'package';
  submitted_at: string;
  last_updated_at: string;
  notes?: string;
  job?: import('./job').Job;
}

export interface ApplicationStats {
  total_applied: number;
  interviews: number;
  offers: number;
  rejection_rate: number;
}
